src/bech32.cpp

```diff
@@ -397,12 +398,12 @@ DecodeResult Decode(const std::string& str) {
 }
 
 /** Find index of an incorrect character in a Bech32 string. */
-std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str) {
+std::pair<std::string, std::vector<int>> LocateErrors(const std::string& str, CharLimit limit) {
     std::vector<int> error_locations{};
 
-    if (str.size() > 90) {
```

Comment:
I worked around this 90 in my pr in https://github.com/bitcoin/bitcoin/pull/29607/files#diff-f146300624c06d2e08aadf500952294148a1785edd6ff2e8b50f13b2c08255edL315 (though I'm not sure why it's not replaced here with `limit`) - I will try to review this in more detail next week

Reply:
Ah, can't believe I missed that. Nice catch!

Reply:
ctrl+f "90", whew that's the only one. This is exactly why named constants are so important to name. I think it's fair to call this a nice cleanup (getting rid of the constant) in addition to the new feature.

Reply:
Oof, also can't believe I missed this. Thanks for taking a look @paplorinc ! Took a look at your PR and I think this is the most elegant way to solve getting rid of the 90 in ExpandHRP, so I cherry-picked your last commit in (added `CHECKSUM_SIZE` and modified the commit message to fit this PR a bit more). As a nice side benefit, just this last commit improves the benchmark.

Reply:
Awesome

Reply:
Awesome!
I just noticed my email was wrong there, would you be so kind and change it to `Lőrinc <pap.lorinc@gmail.com>`?

Reply:
Is this a leftover limit https://github.com/bitcoin/bitcoin/blob/master/src/test/fuzz/bech32.cpp#L32, or it's fine to hard-code in tests?

Reply:
Good catch, this is just a leftover limit. Opened https://github.com/bitcoin/bitcoin/pull/30596 as a follow-up fix.